file_store/src/settings.rs

```diff
@@ -35,4 +25,13 @@ impl Settings {
             .and_then(|config| config.try_deserialize())
             .map_err(Error::from)
     }
+
+    pub async fn connect(&self) -> crate::Client {
+        crate::new_client(
+            self.endpoint.clone(),
+            self.access_key_id.clone(),
+            self.secret_access_key.clone(),
+        )
```

Comment:
Our production happy path being to call this function where we expect all the settings to not be provided makes me feel like we're missing an abstraction. I'm not sure what. But I'll be thinking.

Reply:
it is a fair point, in our production settings we inherit all settings to connect to s3 from the environment.  These properties are only necessary when doing local testing.